### PR TITLE
feat: adds branch plugin

### PR DESCRIPTION
**Pull request by konoufo**

### CODE REVIEW

**Review by oscb — CHANGES_REQUESTED** [review]
Just left a few questions, in particular about the `//ts-ignores` that are throughout the files which we should remove

File: packages/plugins/plugin-branch/src/methods/__tests__/alias.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//@ts-ignore
```
[the remark this answers is not in the document]
Comment (konoufo) [reply packages/plugins/plugin-branch/src/methods/__tests__/alias.test.ts]: nah you're right; red lines in vscode were just hurting my eyes

File: packages/plugins/plugin-branch/src/parameterMapping.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  variant: 'productVariant',
+  quantity: 'quantity',
+  price: 'price',
+  image_url: 'contentImageUrl',
+  url: 'canonicalUrl',
+};
+
+export const transformMap: { [key: string]: (value: any) => any } = {
+  event: (value: string) => {
+    if (value in mapEventNames) {
```
[the remark this answers is not in the document]
Comment (konoufo) [reply packages/plugins/plugin-branch/src/parameterMapping.ts]: the event names ? or the keys ?

[the remark this answers is not in the document]
Comment (konoufo) [reply packages/plugins/plugin-branch/src/parameterMapping.ts]: Do you mean like `if (value.capitalize() in mapEventNames)` ?

File: packages/plugins/plugin-branch/src/BranchPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import {
+  DestinationPlugin,
+  PluginType,
+  TrackEventType,
+  ScreenEventType,
+  IdentifyEventType,
+  AliasEventType,
+
+  // @ts-ignore
```
[the remark this answers is not in the document]
Comment (konoufo) [reply packages/plugins/plugin-branch/src/BranchPlugin.ts]: Indeed these were useless. Just vscode being clueless and complaining.

**Review by oscb — APPROVED** [review]

**Pull request merged**